### PR TITLE
rework consumers

### PR DESCRIPTION
**Pull request by BelehovEgor**

### CODE REVIEW

**Review by VLiamin — APPROVED** [review]

**Review by MariyaSin — APPROVED** [review]

File: src/ProjectService.Data/ProjectRepository.cs [hunk 1]
```diff
@@ -164,28 +165,55 @@ public List<DbProject> Find(List<Guid> projectIds)
...
+            totalCount = projects.Count();
+
+            if (request.SkipCount.HasValue)
+            {
+                projects = projects.Skip(request.SkipCount.Value);
+            }
+
+            if (request.TakeCount.HasValue)
+            {
+                projects = projects.Skip(request.TakeCount.Value);
```
Comment (djino97) [inline src/ProjectService.Data/ProjectRepository.cs:208]: projects.TakeCount

File: src/ProjectService.Data/ProjectRepository.cs [hunk 2]
```diff
@@ -164,28 +165,55 @@ public List<DbProject> Find(List<Guid> projectIds)
...
 
-            return response;
+            if (request.ProjectsIds != null && request.ProjectsIds.Any())
+            {
+                projects = projects.Where(p => request.ProjectsIds.Contains(p.Id));
+            }
+
+            if (request.DepartmentId.HasValue)
+            {
+                projects = projects.Where(p => p.DepartmentId == request.DepartmentId);
```
Comment (djino97) [inline src/ProjectService.Data/ProjectRepository.cs:196]: request.DepartmentId.Value

File: src/ProjectService.Data/ProjectRepository.cs [hunk 3]
```diff
@@ -164,28 +165,55 @@ public List<DbProject> Find(List<Guid> projectIds)
...
+            {
+                projects = projects.Skip(request.SkipCount.Value);
+            }
+
+            if (request.TakeCount.HasValue)
+            {
+                projects = projects.Skip(request.TakeCount.Value);
+            }
+
+            if (request.IncludeUsers && !request.UserId.HasValue)
```
Comment (djino97) [inline src/ProjectService.Data/ProjectRepository.cs:211]: check first request.IncludeUsers.HasValue

File: src/ProjectService.Data/UserRepository.cs [hunk 4]
```diff
@@ -119,5 +119,63 @@ public bool AreExist(params Guid[] ids)
...
+                users = users.Take(takeCount.Value);
+            }
+
+            return users.ToList();
+        }
+
+        public List<DbProjectUser> Get(Guid projectId, int? skipCount, int? takeCount, out int totalCount)
+        {
+            throw new NotImplementedException();
+        }
```
Comment (djino97) [inline src/ProjectService.Data/UserRepository.cs:145]: This necessary?

**Review by Cheephot — APPROVED** [review]

File: src/ProjectService.Data/ProjectRepository.cs (hunk 1 above)
Comment (BelehovEgor) [reply src/ProjectService.Data/ProjectRepository.cs:208]: fixed

File: src/ProjectService.Data/ProjectRepository.cs (hunk 2 above)
Comment (BelehovEgor) [reply src/ProjectService.Data/ProjectRepository.cs:196]: fixed

File: src/ProjectService.Data/ProjectRepository.cs (hunk 3 above)
Comment (BelehovEgor) [reply src/ProjectService.Data/ProjectRepository.cs:211]: it is not nullable

File: src/ProjectService.Data/UserRepository.cs (hunk 4 above)
Comment (BelehovEgor) [reply src/ProjectService.Data/UserRepository.cs]: removed

**Review by ndoriomedov — APPROVED** [review]

**Pull request merged**